earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -329,12 +416,33 @@ ReadDocumentFromString(khxml::DOMLSParser *parser,
 bool
 DestroyDocument(khxml::DOMDocument *doc) throw()
 {
+  bool retval = false;
   try {
     doc->release();
-    return true;
+    retval = true;
   } catch (...) {
   }
-  return false;
+  if (doTerminate)
+  {
+    XMLPlatformUtils::Terminate();
+    try {
+      try {
+         validateXMLParameters();
```

Comment:
I don't see a need to validate the parameters every time we use them. As long as we validate them on input we should be fine.